runelite-client/src/main/java/net/runelite/client/ui/components/DragAndDropReorderPane.java

```diff
@@ -0,0 +1,203 @@
+		if (draggingComponent == null)
+		{
+			dragStartPoint = null;
+			return;
+		}
+		dragYOffset = SwingUtilities.convertPoint(this, dragStartPoint, draggingComponent).y;
+		dragIndex = getPosition(draggingComponent);
+		setLayer(draggingComponent, DRAG_LAYER);
+		moveDraggingComponent(point);
+	}
+
+	private void drag(Point point)
+	{
+		moveDraggingComponent(point);
+
+		// reorder components overlapping with the dragging components mid-point
+		Point draggingComponentMidPoint = SwingUtilities.convertPoint(
+			draggingComponent,
+			new Point(draggingComponent.getWidth() / 2, draggingComponent.getHeight() / 2),
+			this
+		);
+		Component component = getDefaultLayerComponentAt(draggingComponentMidPoint);
+		if (component != null)
+		{
+			int index = getPosition(component);
+			dragIndex = index < dragIndex ? index : index + 1;
+			revalidate();
+		}
+	}
+
+	private void finishDragging()
+	{
+		if (draggingComponent != null)
+		{
+			setLayer(draggingComponent, DEFAULT_LAYER, dragIndex);
+			draggingComponent = null;
+			dragYOffset = 0;
+			dragIndex = -1;
+			revalidate();
+		}
+		dragStartPoint = null;
+	}
+
+	private void moveDraggingComponent(Point point)
+	{
+		// shift the dragging component to match it's earlier y offset with the mouse
+		int y = point.y - dragYOffset;
+		// clamp the height to stay within the pane
+		y = Math.max(y, 0);
+		y = Math.min(y, getHeight() - draggingComponent.getHeight());
+
+		draggingComponent.setLocation(new Point(0, y));
+	}
+
+	private Component getDefaultLayerComponentAt(Point point)
+	{
+		for (Component component : getComponentsInLayer(DEFAULT_LAYER))
+		{
+			if (component.contains(point.x - component.getX(), point.y - component.getY()))
+			{
+				return component;
+			}
+		}
+		return null;
+	}
+
+	private class DragAndDropReorderLayoutManager extends BoxLayout
+	{
+		private DragAndDropReorderLayoutManager()
+		{
+			super(DragAndDropReorderPane.this, BoxLayout.Y_AXIS);
+		}
+
+		@Override
+		public void layoutContainer(Container target)
+		{
+			if (draggingComponent != null)
+			{
+				// temporarily move the dragging component to the default layer for correct layout calculation
+				Point location = draggingComponent.getLocation();
+				setLayer(draggingComponent, DEFAULT_LAYER, dragIndex);
+				super.layoutContainer(target);
+				setLayer(draggingComponent, DRAG_LAYER);
+				draggingComponent.setLocation(location);
+			}
+			else
+			{
+				super.layoutContainer(target);
+			}
+		}
+	}
+
+	private class DragAndDropReorderMouseAdapter extends MouseAdapter
+	{
+		@Override
+		public void mousePressed(MouseEvent e)
+		{
+			if (SwingUtilities.isLeftMouseButton(e) && getComponentCount() > 1)
+			{
+				dragStartPoint = e.getPoint();
+			}
+		}
+
+		@Override
+		public void mouseDragged(MouseEvent e)
+		{
+			if (SwingUtilities.isLeftMouseButton(e) && dragStartPoint != null)
+			{
+				Point point = e.getPoint();
+				if (!contains(point))
+				{
+					finishDragging();
```

Comment:
Imo this should not stop dragging when OOB. Mainly when OOB means the size of parent container what is size of the actual listed xp trackers, so the dragging just stops there and it is pretty annoying.

Reply:
Yeah that makes sense. Updated.
![sample4](https://user-images.githubusercontent.com/11037113/83957601-3f269580-a8bd-11ea-9c67-1fe28b5ddb5b.gif)